src/ua_types_encoding_binary.c

```diff
@@ -778,8 +786,15 @@ ExtensionObject_decodeBinary(UA_ExtensionObject *dst, const UA_DataType *_) {
         /* helping clang analyzer, typeId is numeric */
         UA_assert(typeId.identifier.byteString.data == NULL);
         UA_assert(typeId.identifier.string.data == NULL);
-        typeId.identifier.numeric -= UA_ENCODINGOFFSET_BINARY;
-        findDataType(&typeId, &type);
+        size_t i;
+        for (i = 0; i<UA_TYPES_COUNT; i++) {
```

Comment:
Part 6 says:
> The identifier for the DataTypeEncoding node in the Server's AddressSpace.
ExtensionObjects defined by the OPC UA specification have a numeric node
identifier assigned to them with a NamespaceIndex of 0. The numeric
identifiers are defined in A.3.

As I understand the text, we should compare for the binary encoding identifier as we do in the other two places.
Please correct me if I missunderstood.